tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,1923 @@
+        /// </summary>
+        private string GetConnectionString()
+        {
+            return _connectionString;
+        }
+
+        /// <summary>
+        /// Creates a unique prefix for test keys to ensure test isolation
+        /// </summary>
+        private string GetUniqueKeyPrefix(string testName)
+        {
+            // Use a combination of the test prefix and test method name to ensure uniqueness
+            return $"{TestKeyPrefix}-{testName}-{Guid.NewGuid().ToString("N").Substring(0, 8)}";
+        }
+
+        /// <summary>
+        /// Creates a snapshot with the given name containing the test context's settings
+        /// </summary>
+        private async Task<string> CreateSnapshot(string snapshotName, TestContext testContext)
+        {
+            // Create a snapshot with the test keys
+            var settingsToInclude = new List<ConfigurationSettingsFilter>
+            {
+                new ConfigurationSettingsFilter($"{testContext.KeyPrefix}:*")
+            };
+
+            ConfigurationSnapshot snapshot = new ConfigurationSnapshot(settingsToInclude);
+
+            snapshot.SnapshotComposition = SnapshotComposition.Key;
+
+            try
+            {
+                // Create the snapshot
+                CreateSnapshotOperation operation = await _configClient.CreateSnapshotAsync(WaitUntil.Completed, snapshotName, snapshot);
+
+                // Track created snapshot for cleanup
+                _createdSnapshotNames.Add(snapshotName);
+
+                return operation.Value.Name;
+            }
+            catch (RequestFailedException ex)
+            {
+                Console.WriteLine($"Error creating snapshot: {ex.Message}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Initialize clients to connect to the existing App Configuration store and Key Vault.
+        /// </summary>
+        public async Task InitializeAsync()
+        {
+            try
+            {
+                var credential = GetCredential();
+                string subscriptionId = GetCurrentSubscriptionId();
+
+                // Initialize Azure Resource Manager client
+                var armClient = new ArmClient(credential);
+                SubscriptionResource subscription = armClient.GetSubscriptionResource(new ResourceIdentifier($"/subscriptions/{subscriptionId}"));
+
+                ResourceGroupResource resourceGroup = await subscription.GetResourceGroups().GetAsync(ResourceGroupName);
+
+                AppConfigurationStoreResource appConfigStore = null;
+                KeyVaultResource keyVault = null;
+
+                try
+                {
+                    // Get App Configuration store directly using the resource group and store name
+                    appConfigStore = await resourceGroup.GetAppConfigurationStores().GetAsync(AppConfigStoreName);
+                }
+                catch (RequestFailedException ex)
```

Comment:
The error message expects a not found failure, but you're catching all request failed exceptions.